src/wp-includes/script-loader.php

```diff
@@ -2607,6 +2605,46 @@ function wp_should_load_separate_core_block_assets() {
 	return apply_filters( 'should_load_separate_core_block_assets', false );
 }
 
+/**
+ * Checks whether block styles should be loaded only on-render.
+ *
+ * When this function returns true, other functions ensure that blocks only load their assets on-render.
+ * When this function returns false, all block assets are loaded regardless of whether they are rendered in a page.
+ *
+ * The default return value depends on the result of {@see wp_should_load_separate_core_block_assets()}, which controls
+ * whether Core block stylesheets should be loaded separately or via a combined 'wp-block-library' stylesheet.
+ *
+ * This only affects front end and not the block editor screens.
+ *
+ * @since 6.8.0
+ * @see wp_should_load_separate_core_block_assets()
+ *
+ * @return bool Whether to load block assets only when they are rendered.
+ */
+function wp_should_load_block_assets_on_demand() {
+	if ( is_admin() || is_feed() || wp_is_rest_endpoint() ) {
+		return false;
+	}
+
+	$load_assets_on_demand = wp_should_load_separate_core_block_assets();
```

Comment:
```suggestion
	/**
	 * The default return value depends on the result of {@see wp_should_load_separate_core_block_assets()}, which controls
	 * whether Core block stylesheets should be loaded separately or via a combined 'wp-block-library' stylesheet.
	 */
	$load_assets_on_demand = wp_should_load_separate_core_block_assets();
```

I see that the reason why we default is listed in the main doc comment of the function. But I feel it would be valuable to have inline

Reply:
Good point. I would amend the text though to explain that this is due to backward compatibility - in case anyone wonders how the two are related.

Reply:
Updated in https://github.com/WordPress/wordpress-develop/pull/8234/commits/dc2944472bfd01212765cc2f9a62b921b2b1482a, LMKWYT.

Reply:
Thats great 👍